include/exadg/poisson/spatial_discretization/operator.cpp

```diff
@@ -226,6 +226,24 @@ Operator<dim, Number, n_components>::setup(
   matrix_free      = matrix_free_in;
   matrix_free_data = matrix_free_data_in;
 
+  if(not(boundary_descriptor->dirichlet_cached_bc.empty()))
+  {
+    interface_data_dirichlet_cached =
+      std::make_shared<ContainerInterfaceData<dim, n_components, Number>>();
+    std::vector<unsigned int> quad_indices;
+    if(param.spatial_discretization == SpatialDiscretization::DG)
+      quad_indices.emplace_back(get_quad_index());
+    else if(param.spatial_discretization == SpatialDiscretization::CG)
+      quad_indices.emplace_back(get_quad_index_gauss_lobatto());
+    else
+      AssertThrow(false, dealii::ExcMessage("not implemented."));
```

Comment:
What about:
```suggestion
      AssertThrow(false, dealii::ExcNotImplemented());
```

Reply:
Yes, would probably make sense. As for other comments, I would prefer to do this consistently within ExaDG due to many occurrences.
As a very general comment or question: I am not sure if application projects such as ExaDG should use  `Assert`/`AssertThrow` (from dealii) everywhere, or instead (re-)define assert-functionality within the library? One problem that I see is for example that `Assert`/`AssertThrow` does not have the `dealii` namespace, see PR #145.

Reply:
We can create our own exceptions. That shouldn't be a problem. Let me create PR tomorrow for that!

Reply:
Regarding `Assert`: That is a macro and thus gets resolved by the preprocessor. One can avoid the use of those, see e.g. the comment https://github.com/dealii/dealii/blob/09bae93a06eca37f12e27be378647160377a63a1/include/deal.II/base/exceptions.h#L1434-L1492 . When it comes to defining our own macros instead, the driving force is likely the added value we get from having a new definition, as the current variant is pretty self-explaining and I don't see fundamental problems with the current usage.

Reply:
Yes, I had this comment in mind. Might this raise problems in the future? Why using macros and not something more modern, or do you think the macro-style is still the best solution nowadays?

The question is whether we should introduce dependencies on deal.II in files/modules, that actually do not depend on deal.II (apart from exceptions). Even though ExaDG will probably always have deal.II as a mandatory dependency, I think it would still make sense to not copy such functionality from deal.II.

Reply:
I don't expect problems in the regular use case by the macro, but it of course depends on if there is a conflicting definition of the macro in another package, or if one wants to redefine it to something else (this includes the `DeclExceptionMsg` and friends). One can disable those flags and keep separate.

The question for us in ExaDG is whether it is worth the effort to design something new, i.e., we have the competence to design something substantially better and if we want to spend the developer time on this topic rather than some other topic.

> Why using macros and not something more modern, or do you think the macro-style is still the best solution nowadays?

I am not an expert on this topic and do not intend to become one, so I can't really say much beyond a personal opinion: For now I see little downsides with this approach, it is robust, makes it pretty simple to assert things or create new exceptions and without an unreasonably high learning curve. I would act if it does not fit our needs any more to the extent that creating a solution for us consumes less aggregate time than fighting with undesired behavior. I can't see that now. Any way, I am happy to learn about alternatives that are as easy or easier to use, and you would consider "more modern".